Lib/threading.py

```diff
@@ -1027,6 +1032,8 @@ def _bootstrap_inner(self):
             self._set_ident()
             if _HAVE_THREAD_NATIVE_ID:
                 self._set_native_id()
+            if _HAVE_SET_NAME and self._name:
+                _set_name(self._name)
```

Comment:
`_HAVE_SET_NAME` seems unnecessary; did you consider instead setting `_set_name = None` if it's not available?
```suggestion
            if _set_name is non None and self._name:
                _set_name(self._name)
```

There's no C macro like `PY_HAVE_THREAD_NATIVE_ID`, so it seems better to be consistent with `_CRLock` than with `get_native_id`.